internal/command/sms_config.go

```diff
@@ -110,10 +110,87 @@ func (c *Commands) ChangeSMSConfigTwilioToken(ctx context.Context, instanceID, i
 	return writeModelToObjectDetails(&smsConfigWriteModel.WriteModel), nil
 }
 
-func (c *Commands) ActivateSMSConfig(ctx context.Context, instanceID, id string) (*domain.ObjectDetails, error) {
+type AddSMSHTTP struct {
+	Endpoint string
+}
+
+func (c *Commands) AddSMSConfigHTTP(ctx context.Context, instanceID string, config *AddSMSHTTP) (string, *domain.ObjectDetails, error) {
+	id, err := c.idGenerator.Next()
+	if err != nil {
+		return "", nil, err
+	}
+	smsConfigWriteModel, err := c.getSMSConfig(ctx, instanceID, id)
+	if err != nil {
+		return "", nil, err
+	}
+
+	iamAgg := InstanceAggregateFromWriteModel(&smsConfigWriteModel.WriteModel)
+	pushedEvents, err := c.eventstore.Push(ctx, instance.NewSMSConfigHTTPAddedEvent(
+		ctx,
+		iamAgg,
+		id,
+		config.Endpoint))
+	if err != nil {
+		return "", nil, err
+	}
+	err = AppendAndReduce(smsConfigWriteModel, pushedEvents...)
+	if err != nil {
+		return "", nil, err
+	}
+	return id, writeModelToObjectDetails(&smsConfigWriteModel.WriteModel), nil
+}
+
+type ChangeSMSHTTP struct {
+	Endpoint string
+}
+
+func (c *Commands) ChangeSMSConfigHTTP(ctx context.Context, instanceID, id string, config *ChangeSMSHTTP) (*domain.ObjectDetails, error) {
+	if id == "" {
+		return nil, zerrors.ThrowInvalidArgument(nil, "SMS-e9jwf", "Errors.IDMissing")
+	}
+	smsConfigWriteModel, err := c.getSMSConfig(ctx, instanceID, id)
+	if err != nil {
+		return nil, err
+	}
+	if !smsConfigWriteModel.State.Exists() || smsConfigWriteModel.HTTP == nil {
+		return nil, zerrors.ThrowNotFound(nil, "COMMAND-2m9fw", "Errors.SMSConfig.NotFound")
+	}
+	iamAgg := InstanceAggregateFromWriteModel(&smsConfigWriteModel.WriteModel)
+
+	changedEvent, hasChanged, err := smsConfigWriteModel.NewHTTPChangedEvent(
+		ctx,
+		iamAgg,
+		id,
+		config.Endpoint)
+	if err != nil {
+		return nil, err
+	}
+	if !hasChanged {
+		return nil, zerrors.ThrowPreconditionFailed(nil, "COMMAND-jf9wk", "Errors.NoChangesFound")
+	}
+	pushedEvents, err := c.eventstore.Push(ctx, changedEvent)
+	if err != nil {
+		return nil, err
+	}
+	err = AppendAndReduce(smsConfigWriteModel, pushedEvents...)
+	if err != nil {
+		return nil, err
+	}
+	return writeModelToObjectDetails(&smsConfigWriteModel.WriteModel), nil
+}
+
+func (c *Commands) ActivateSMSConfig(ctx context.Context, instanceID, id, activatedId string) (*domain.ObjectDetails, error) {
 	if id == "" {
 		return nil, zerrors.ThrowInvalidArgument(nil, "SMS-dn93n", "Errors.IDMissing")
 	}
+
+	if len(activatedId) > 0 {
+		_, err := c.DeactivateSMSConfig(ctx, instanceID, activatedId)
+		if err != nil {
+			return nil, err
+		}
+	}
```

Comment:
the command does not even need to care that the current id is deactivated
the projection and if needed the writemodel just must compute the current active model based on the last activated id, this would allow the command to not create an event if the current id is tried to be activated again
we then would not even need a deactivate event anymore (also not in the api)
BWT why is there a length check on string?